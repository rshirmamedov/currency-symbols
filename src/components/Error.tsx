type ErrorProps = {
    message?: string;
};

export const Error = ({ message }: ErrorProps) => (
    <div className="text-sm text-red-600">{message || "Something went wrong!"}</div>
);
