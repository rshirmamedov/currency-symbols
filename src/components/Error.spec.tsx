import { render } from "@testing-library/react";
import { Error } from "./Error";

describe("Error", () => {
    it("renders with default message", () => {
        const { container } = render(<Error />);

        expect(container).toMatchSnapshot();
    });

    it("renders with custom message", () => {
        const { container } = render(<Error message="Username is required" />);

        expect(container).toMatchSnapshot();
    });
});