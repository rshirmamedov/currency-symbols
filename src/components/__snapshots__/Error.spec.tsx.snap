// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error renders with custom message 1`] = `
<div>
  <div
    class="text-sm text-red-600"
  >
    Username is required
  </div>
</div>
`;

exports[`Error renders with default message 1`] = `
<div>
  <div
    class="text-sm text-red-600"
  >
    Something went wrong!
  </div>
</div>
`;
